### ISSUE TITLE
[RHEL 6.9] Error booting into menu entry created by boom

### ISSUE THREAD

**Issue by bertinatto** [bmr-cymru/boom#11]
Steps:

**1. Make sure Grub 1 support is enabled in Grub 1:**

```
# cat /boot/boom/boom.conf
 
(...)
[legacy]
enable = True
format = grub1
sync = True
```
 
**2. Check snapshot volume:**

```
# lvdisplay | fpaste
Uploading (2.2KiB)...
```
https://paste.fedoraproject.org/paste/bv0LbSwZ8KWl7ya2x8SMpQ/raw

**3. Confirm if */boot* is mounted in a separate partition:**

```
# mount             
/dev/mapper/VolGroup-lv_root on / type ext4 (rw)
proc on /proc type proc (rw)
sysfs on /sys type sysfs (rw)
devpts on /dev/pts type devpts (rw,gid=5,mode=620)
tmpfs on /dev/shm type tmpfs (rw,rootcontext="system_u:object_r:tmpfs_t:s0")
/dev/vda1 on /boot type ext4 (rw)
none on /proc/sys/fs/binfmt_misc type binfmt_misc (rw)
```

**4. List profiles:**

```
# boom profile list
OsID    Name                            OsVersion               
9cb53dd Fedora                          24 (Workstation Edition)
9736c34 Fedora                          25 (Server Edition)     
d4439b7 Fedora                          26 (Workstation Edition)
efd6d41 Red Hat Enterprise Linux Server 4 (Server)              
98c3edb Red Hat Enterprise Linux Server 6 (Server)              
c0b921e Red Hat Enterprise Linux Server 7 (Server)              
3fc389b Red Hat Enterprise Linux Server 7.2 (Maipo)             
21e37c8 Ubuntu                          16.04 LTS (Xenial Xerus)
```

**5. Create boot entry for the snapshot:**

```
# boom create --profile 98c3edb --title "REHEL Snapshot" --root-lv VolGroup/snap_root
WARNING - No Grub2 configuration file found
WARNING - Boom configuration not found in grub.cfg
WARNING - Run 'grub2-mkconfig > /boot/grub2/grub.cfg' to enable
Probing devices to guess BIOS drives. This may take a long time.
Created entry with boot_id bf02672:
  title REHEL Snapshot
  machine-id cb24bb95ae188d735bcf17db00000006
  version 2.6.32-696.23.1.el6.x86_64
  linux /vmlinuz-2.6.32-696.23.1.el6.x86_64
  initrd /initramfs-2.6.32-696.23.1.el6.x86_64.img
  options root=/dev/VolGroup/snap_root ro rd.lvm.lv=VolGroup/snap_root rhgb quiet

```

**6. Check if menu entry was created:**

```
# cat /boot/grub/grub.conf 
# grub.conf generated by anaconda
#
# Note that you do not have to rerun grub after making changes to this file
# NOTICE:  You have a /boot partition.  This means that
#          all kernel and initrd paths are relative to /boot/, eg.
#          root (hd0,0)
#          kernel /vmlinuz-version ro root=/dev/mapper/VolGroup-lv_root
#          initrd /initrd-[generic-]version.img
#boot=/dev/vda
default=0
timeout=5
splashimage=(hd0,0)/grub/splash.xpm.gz
hiddenmenu
title Red Hat Enterprise Linux Server (2.6.32-696.23.1.el6.x86_64)
	root (hd0,0)
	kernel /vmlinuz-2.6.32-696.23.1.el6.x86_64 ro root=/dev/mapper/VolGroup-lv_root rd_NO_LUKS LANG=en_US.UTF-8 rd_NO_MD rd_LVM_LV=VolGroup/LogVol02 SYSFONT=latarcyrheb-sun16 crashkernel=auto rd_LVM_LV=VolGroup/lv_root  KEYBOARDTYPE=pc KEYTABLE=us rd_NO_DM rhgb quiet
	initrd /initramfs-2.6.32-696.23.1.el6.x86_64.img
title Red Hat Enterprise Linux 6 (2.6.32-696.el6.x86_64)
	root (hd0,0)
	kernel /vmlinuz-2.6.32-696.el6.x86_64 ro root=/dev/mapper/VolGroup-lv_root rd_NO_LUKS LANG=en_US.UTF-8 rd_NO_MD rd_LVM_LV=VolGroup/LogVol02 SYSFONT=latarcyrheb-sun16 crashkernel=auto rd_LVM_LV=VolGroup/lv_root  KEYBOARDTYPE=pc KEYTABLE=us rd_NO_DM rhgb quiet
	initrd /initramfs-2.6.32-696.el6.x86_64.img
#--- BOOM_Grub1_BEGIN ---
title REHEL Snapshot
        root None
        kernel /vmlinuz-2.6.32-696.23.1.el6.x86_64 root=/dev/VolGroup/snap_root ro rd.lvm.lv=VolGroup/snap_root rhgb quiet
        initrd /initramfs-2.6.32-696.23.1.el6.x86_64.img
#--- BOOM_Grub1_END ---

```

**7. Reboot and selected the boot entry recently created by boom:**

https://i.imgur.com/SEwbOaQ.png

**Comment by bmr-cymru** [comment]
Ah - sorry - I spotted this the other day & should have warned you. The built-in profile for RHEL6 is wrong at the moment. It uses the new Dracut syntax rather than the old style that is correct for RHEL6.

If you edit the profile file (using boom or an editor) to have this content:

```
BOOM_OS_ID="98c3edb94b7b3c8c95cb7d93f75693d2b25f764d"
BOOM_OS_NAME="Red Hat Enterprise Linux Server"
BOOM_OS_SHORT_NAME="rhel"
BOOM_OS_VERSION="6 (Server)"
BOOM_OS_VERSION_ID="6"
BOOM_OS_UNAME_PATTERN="el6"
BOOM_OS_KERNEL_PATTERN="/vmlinuz-%{version}"
BOOM_OS_INITRAMFS_PATTERN="/initramfs-%{version}.img"
BOOM_OS_ROOT_OPTS_LVM2="rd_LVM_LV=%{lvm_root_lv}"
BOOM_OS_ROOT_OPTS_BTRFS="rootflags=%{btrfs_subvolume}"
BOOM_OS_OPTIONS="root=%{root_device} ro %{root_opts} rhgb quiet"
```

The `BOOM_OS_ROOT_OPTS_LVM2` key needs to use  `rd_LVM_LV` instead of `rd.lvm.lv`.

**Comment by bertinatto** [comment]
Thanks for looking into this, @bmr-cymru.

I changed the profile as you suggested and re-created the entry with the new profile (just in case), but still I got that error.

**Comment by bmr-cymru** [comment]
Ah, I think you are hitting another bug I noticed last week. I didn't spot it as I noticed the incorrect Dracut params first. This one is fixed in master, but not yet in the Copr builds:

```
commit bb7ad0797e691fd574bc99a46ea2b62d98b721e9
Author: Bryn M. Reeves <bmr@redhat.com>
Date:   Fri Mar 16 16:41:59 2018 +0000

    boom: fix grub1 command and output parsing
    
    Fixes: #10
    
    Signed-off-by: Bryn M. Reeves <bmr@redhat.com>
```

The patch is pretty simple but I'll kick off a new build in a moment:

```diff
diff --git a/boom/legacy.py b/boom/legacy.py
index 792762a..0d765d8 100644
--- a/boom/legacy.py
+++ b/boom/legacy.py
@@ -69,7 +69,7 @@ def _get_grub1_device(force=False):
     # The grub1 binary
     grub_cmd = "grub"
     # The command to issue to discover the /boot device
-    find_cmd = "find " + _loader_map[BOOM_LOADER_GRUB1][2]
+    find_cmd = "find /%s\n" % _loader_map[BOOM_LOADER_GRUB1][2]
     # Regular expression matching a valid grub device string
     find_rgx = r" \(hd\d+,\d+\)"
 
@@ -80,11 +80,10 @@ def _get_grub1_device(force=False):
     except OSError:
         raise BoomLegacyLoaderError("Could not execute grub1 shell.")
 
-    out = out[0]
-
-    for line in out:
+    for line in out[0].splitlines():
         if re.match(find_rgx, line):
             __grub1_device = line.lstrip().rstrip()
+            _log_debug("Set grub1 device to '%s'" % __grub1_device)
             return _get_grub1_device()
 
 
```

**Comment by bertinatto** [comment]
Update from last Copr build and it worked.

Thanks for the quick fix, @bmr-cymru!
